azurerm/internal/services/apimanagement/api_management_resource.go

```diff
@@ -820,6 +829,11 @@ func flattenApiManagementHostnameConfigurations(input *[]apimanagement.HostnameC
 			output["host_name"] = *config.HostName
 		}
 
+		// There'll always be a default custom domain with hostName "apim_name.azure-api.net" and Type "Proxy", which should be ignored
+		if *config.HostName == strings.ToLower(name)+".azure-api.net" && config.Type == apimanagement.HostnameTypeProxy {
```

Comment:
`azure-api.net` is only valid in Azure Public - looks like this'll need to be fetched/added to the [environments list in Azure/go-autorest](https://github.com/Azure/go-autorest/blob/master/autorest/azure/environments.go#L91-L99)

Reply:
Upstream issue opened: https://github.com/Azure/azure-rest-api-specs/issues/10065